File: objective-c/The Logic/StartViewController.m
//
//  ViewController.m
//  The Logic
//
//  Created by babi4_97 on 05.11.2017.
//  Copyright © 2017 Ivan Babkin. All rights reserved.
//

#import "StartViewController.h"
#import "GameConst.h"

@interface StartViewController () {
    
    __weak IBOutlet UIView *_hintView;
    __weak IBOutlet UISwitch *_hintSwitch;
}

@end

@implementation StartViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    if (![[NSUserDefaults standardUserDefaults] stringForKey:kGameConstRules]) {
        _hintView.hidden = YES;
    }
    else {
        _hintSwitch.on = [[NSUserDefaults standardUserDefaults] boolForKey:kGameConstHint];
    }
}

- (IBAction)hintsSwitched:(id)sender {
    [[NSUserDefaults standardUserDefaults] setBool:_hintSwitch.on forKey:kGameConstHint];
}

- (IBAction)startGame:(id)sender {
}

- (IBAction)rules:(id)sender {
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: swift/The logic/StartViewController.swift
//
//  ViewController.swift
//  The logic
//
//  Created by Ivan Babkin on 12.06.2018.
//  Copyright © 2018 Ivan Babkin. All rights reserved.
//

import UIKit

class StartViewController: UIViewController {

    @IBOutlet weak var hintView: UIView!
    @IBOutlet weak var hintSwitch: UISwitch!
    
    override func viewWillAppear(_ animated: Bool) {
        let ud = UserDefaults.standard
        if ud.string(forKey: Constants.kRules) != nil {
            hintView.isHidden = true
        }
        else {
            hintSwitch.isOn = ud.bool(forKey: Constants.kHint)
        }
    }
    
    @IBAction func hintSwitched(_ sender: UISwitch) {
        let ud = UserDefaults.standard
        ud.set(hintSwitch.isOn, forKey: Constants.kHint)
    }
    
}

